src/Symfony/Component/Routing/Loader/AnnotationClassLoader.php

```diff
@@ -157,11 +162,95 @@ protected function addRoute(RouteCollection $collection, $annot, $globals, \Refl
         }
 
-        $route = $this->createRoute($globals['path'].$annot->getPath(), $defaults, $requirements, $options, $host, $schemes, $methods, $condition);
+        $path = $annot->getPath();
+        $locales = $annot->getLocales();
+        $hasLocalizedPrefix = false === empty($globals['locales']);
+        $hasPrefix = $hasLocalizedPrefix || false === empty($globals['path']);
+        $isLocalized = !empty($locales);
+        $hasPathOrLocales = false === empty($path) || $isLocalized;
 
-        $this->configureRoute($route, $class, $method, $annot);
+        if (false === $hasPrefix && false === $hasPathOrLocales) {
+            throw new \LogicException("annotation for {$class->name}::{$method->name} has no path.");
+        }
 
-        $collection->add($name, $route);
+        if (!$hasPathOrLocales) {
+            if ($hasLocalizedPrefix) {
+                foreach ($globals['locales'] as $locale => $localePath) {
+                    $routeName = "{$name}.{$locale}";
+                    $route = $this->createRoute($localePath, $defaults, $requirements, $options, $host, $schemes, $methods, $condition);
+                    $this->configureRoute($route, $class, $method, $annot);
+                    $route->setDefault('_locale', $locale);
+                    $collection->add($routeName, $route);
+                }
+            } else {
+                $route = $this->createRoute($globals['path'], $defaults, $requirements, $options, $host, $schemes, $methods, $condition);
+                $this->configureRoute($route, $class, $method, $annot);
+                $collection->add($name, $route);
+            }
+        } elseif (!$hasPrefix) {
+            if ($isLocalized) {
+                foreach ($locales as $locale => $localePath) {
+                    $routeName = "{$name}.{$locale}";
+                    $route = $this->createRoute($localePath, $defaults, $requirements, $options, $host, $schemes, $methods, $condition);
+                    $this->configureRoute($route, $class, $method, $annot);
+                    $route->setDefault('_locale', $locale);
+                    $collection->add($routeName, $route);
+                }
+            } else {
+                $route = $this->createRoute($path, $defaults, $requirements, $options, $host, $schemes, $methods, $condition);
+                $this->configureRoute($route, $class, $method, $annot);
+                $collection->add($name, $route);
+            }
+        } else {
+            if ($hasLocalizedPrefix) {
+                if ($isLocalized) {
+                    $missing = array_diff(array_keys($globals['locales']), array_keys($locales));
+
+                    if (!empty($missing)) {
+                        $missingLocales = implode(' and ', $missing);
+                        throw new \LogicException("Localized prefix(es) {$missingLocales} is/are not defined on {$class->name} while it is/are defined on its action {$method->name}.");
```

Comment:
missing double quotes around var placeholder
we also usually use sprintf in similar situations (same below)